devices/ePaper/Drivers/Ssd1681.cs

```diff
@@ -0,0 +1,718 @@
+        }
+
+        /// <summary>
+        /// Perform the required initialization steps to set up the display.
+        /// </summary>
+        public void Initialize()
+        {
+            if (PowerState != PowerState.PoweredOn)
+                throw new InvalidOperationException("Unable to initialize the display until it has been powered on. Call PowerOn() first.");
+
+            // set gate lines and scanning sequence
+            this.SendCommand((byte)Command.DriverOutputControl);
+            this.SendData((byte)(this.Height - 1), 0x00, 0x00); // refer to spec for description
+
+            // Set data entry sequence
+            this.SendCommand((byte)Command.DataEntryModeSetting);
+            this.SendData(0x03); // Y Increment, X Increment with RAM address counter auto updated in the X direction.
+
+            // Set RAM X start / end position
+            this.SendCommand((byte)Command.SetRAMAddressXStartEndPosition);
+            this.SendData(
+                /* Start at 0*/ 0x00,
+                /* End */ (byte)(this.Width / 8 - 1)); // end at width bits converted to bytes (starts @ 0)
+
+            // Set RAM Y start / end positon
+            this.SendCommand((byte)Command.SetRAMAddressYStartEndPosition);
+            this.SendData(/* Start at 0 */ 0x00, 0x00, /* End */ (byte)(this.Height - 1), 0x00);
+
+            // Set Panel Border
+            this.SendCommand((byte)Command.BorderWaveformControl);
+            this.SendData(0xc0); // was 0x05
+
+            // Set Temperature sensor
+            this.SendCommand((byte)Command.TempSensorControlSelection);
+            this.SendData(0x80); // use internal temp sensor
+
+            // Do a full refresh of the display
+            this.PerformFullRefresh();
+        }
+
+
+        /// <inheritdoc/>
+        public void PerformFullRefresh()
+        {
+            this.SendCommand((byte)Command.BoosterSoftStartControl);
+            this.SendData(0x8b, 0x9c, 0x96, 0x0f);
+
+            this.SendCommand((byte)Command.DisplayUpdateControl2);
+            this.SendData((byte)RefreshMode.FullRefresh); // Display Mode 1 (Full Refresh)
+
+            this.SendCommand((byte)Command.MasterActivation);
+            this.WaitReady();
+        }
+
+        /// <inheritdoc/>
+        public void PerformPartialRefresh()
+        {
+            this.SendCommand((byte)Command.BoosterSoftStartControl);
+            this.SendData(0x8b, 0x9c, 0x96, 0x0f);
+
+            this.SendCommand((byte)Command.DisplayUpdateControl2);
+            this.SendData((byte)RefreshMode.PartialRefresh); // Display Mode 2 (Partial Refresh)
+
+            this.SendCommand((byte)Command.MasterActivation);
+            this.WaitReady();
+        }
+
+
+        /// <inheritdoc/>
+        public void Clear(bool triggerPageRefresh = false)
+        {
+            this.BeginFrameDraw();
+            do
+            {
+                // do nothing. internal frame buffers already cleared by BeginFrameDraw()
```

Comment:
please use a thread.sleep or equivalent so other threads can have a chance to get their turn. Something small like 20 ms will do the job